catalog/dags/maintenance/decode_and_deduplicate_image_tags.py

```diff
@@ -0,0 +1,128 @@
+"""
+See the issue for context and motivation: https://github.com/WordPress/openverse/issues/4452
+
+This DAG triggers a run of the batched update DAG. It generates a new list of tags by
+trimming all existing tags and re-inserting only the distinct tags of the resulting list of tags.
+
+Only records before the CC Search -> Openverse transition are affected. As such, because all
+audio records are dated after that transition, we only need to scan images.
+"""
+
+from datetime import datetime, timedelta
+from textwrap import dedent
+
+from airflow.decorators import dag, task
+from airflow.models.abstractoperator import AbstractOperator
+from airflow.operators.trigger_dagrun import TriggerDagRunOperator
+
+from common.constants import DAG_DEFAULT_ARGS, POSTGRES_CONN_ID
+from common.sql import PostgresHook
+from database.batched_update.constants import DAG_ID as BATCHED_UPDATE_DAG_ID
+
+
+DAG_ID = "decode_and_deduplicate_image_tags"
+
+HAS_RAW_ESCAPED_UNICODE = (
+    r'(@.name like_regex "\\\\(x)([\\da-f]{2})|\\\\(u)([\\da-f]{4})" flag "i")'
+)
+
+
+@task
+def ensure_ov_unistr(
+    postgres_conn_id: str = POSTGRES_CONN_ID,
+    task: AbstractOperator = None,
+):
+    """
+    Create a naïve implementation of Postgres 14+ ``unistr``.
+
+    We are on Postgres 13 and have to do without ``unistr``. For all intents and purposes,
+    this implementation solves the problem for us.
+
+    The ``ov`` prefix prevents clashing with the built-in should we upgrade.
+    """
+
+    postgres = PostgresHook(
+        postgres_conn_id=postgres_conn_id,
+        default_statement_timeout=PostgresHook.get_execution_timeout(task),
+        log_sql=True,
+    )
+
+    return postgres.run(
+        dedent(
+            """
+            CREATE OR REPLACE FUNCTION ov_unistr (string text)
+                RETURNS text
+            AS $$
+                return string.encode().decode("unicode_escape") if string else string
+            $$ LANGUAGE plpython3u;
+            """
+        )
+    )
+
+
+@dag(
+    dag_id=DAG_ID,
+    schedule=None,
+    start_date=datetime(2024, 6, 3),
+    tags=["database"],
+    doc_md=__doc__,
+    max_active_runs=1,
+    default_args=DAG_DEFAULT_ARGS,
+)
+def decode_and_deduplicate_image_tags():
+    ensure_ov_unistr() >> TriggerDagRunOperator(
+        task_id="trigger_batched_update",
+        trigger_dag_id=BATCHED_UPDATE_DAG_ID,
+        wait_for_completion=True,
+        execution_timeout=timedelta(hours=5),
```

Comment:
Are you confident in this timeout? I don't _think_ the triggered dagrun will be timed out if this task times out (eg, I think we'd get a timeout of this DAG but the batched update would continue running just fine) but I'm not actually 100% on that.

Reply:
I am not at all confident in this timeout :sweat_smile: